src/index.js

```diff
@@ -1,24 +1,37 @@
 // Set up your application entry point here...
 ///* eslint-disable import/default */
 
-import React from "react";
-import { render } from "react-dom";
-import { Provider } from "react-redux";
-import { Router, browserHistory } from "react-router";
+import React from 'react';
+import { render } from 'react-dom';
+import { Provider } from 'react-redux';
+import { Router, browserHistory } from 'react-router';
 
-import routes from "./routes";
-import configureStore from "./store/configureStore";
+import routes from './routes';
+import configureStore from './store/configureStore';
+import { UserStateActionDispatcherContext, UserStateContext } from './store/contexts';
+import { userReducer } from './reducers/userReducer';
+import { userStateInitialValue } from './store/states';
 
-import "./styles/styles.scss";
-import { syncHistoryWithStore } from "react-router-redux";
+import './styles/styles.scss';
+import { syncHistoryWithStore } from 'react-router-redux';
 
 const store = configureStore();
 
 // Create an enhanced history that syncs navigation events with the store
 const history = syncHistoryWithStore(browserHistory, store);
 
-render(
-	<Provider store={store}>
-		<Router history={history} routes={routes} />
-	</Provider>, document.getElementById("app")
-);
+const App = () => {
+	const [userState, userActionDispatcher] = React.useReducer(userReducer, userStateInitialValue);
```

Comment:
I don't quite understand this part of logic and this useContext pattern. Maybe you can educate me on that :D

Reply:
Yes sure, let's talk about it over the call.